### DISCUSSION TITLE
Freetz-ng image Erstellung nimmt falsche Fritzbox version

### DISCUSSION THREAD

**Discussion by tutti21** [Freetz-NG/freetz-ng#1085] [Allgemein] [answered]
Moin,

hat noch jemand das Problem wenn man im Freetz-ng tool die Fritzbox 7590 angibt er das Image von der 7690 nimmt siehe Bilder und dann beim Flashversuch kommt natuerlich auch ein Fehler, dass das Image falsch ist.

![1](https://github.com/user-attachments/assets/91d51835-2229-4ac2-b8cd-0f2042bc82ec)
![2](https://github.com/user-attachments/assets/e1cbd91c-d89c-4329-9fe9-8f55e9fac68c)
![3](https://github.com/user-attachments/assets/73a5f4b2-2877-4f66-bc06-33e3a120c017)
![4](https://github.com/user-attachments/assets/eb6c6948-bb6e-4914-81ca-6260bc449d94)

**Comment by fda77** [comment]
Da war doch vor kurzem schonmal einer hier. Ich bin jetzt zu faul zum suchen, vielleicht könntest du das tun
Dass beim flashen eine Warnung kommt weil das image nicht zu hardware passt ist doch okay

**Reply by Conan179** [reply]
https://github.com/Freetz-NG/freetz-ng/discussions/1076
Auffällig, das das image wieder für die 7690 seihen soll...

**Comment by fda77** [comment]
ist 7690 nicht default man man gar nichts auswählt?

**Reply by LizenzFass78851** [reply]
Ja, die 7690 ist default (wenn man keine vorher gespeicherte `.config` hat)
![VirtualBox_Lubuntu_28_12_2024_14_08_00](https://github.com/user-attachments/assets/023b3c1d-932f-4bb4-91a0-005576e2e614)

**edit**:
Wenn man `make` ausführt, ohne vorher über `make menuconfig` eine `.config` zu speichern, nimmt es tatsächlich als Standard die 7690.

**Comment by tutti21** [comment]
![5](https://github.com/user-attachments/assets/0d26de1d-09c5-4d21-a57d-c7a576ef62a5)
![6](https://github.com/user-attachments/assets/ede6173e-243b-45a4-99b2-aa9527051e14)
Es steht ja selbst in der GUI das ich die 7590 ausgewaehlt habe. Ich habe das jetzt 10x versucht mit ner anderen config oder load. 

Ist doch richtig das ich am ende wenn ich mein ausgewaehltes in ein Image schreiben will auf exit gehe richtig? Oder schreibt er dann einfach eine default.. ich hab ka was ich jetzt noch machen soll

**Reply by Conan179** [reply]
Sekunde, deine .config heist 7590.config?
Wo hast du gelesen, die .config müste so heisen?

**Reply by LizenzFass78851** [reply, marked as answer]
Die Config aus `make menuconfig` muss als `.config` gespeichert werden. 
Nicht `7590.config`, sondern nur `.config`.

**Reply by tutti21** [reply]
yep that work sry. Es stand jetzt auf der Seite gar nichts was ich da machen soll und das der default name nicht veraendert werden darf

**Reply by Conan179** [reply]
@tutti21 nach welcher Anleitung bist du gegangen? Wo steht das du es so speichern sollst?

**Reply by tutti21** [reply]
https://github.com/Freetz-NG/freetz-ng/blob/master/README.md#requirements da steht gar nix aber ich bin einfach mal davon ausgegangen das ich etwas speichern muss, wenn ichw as veraendern will.

**Comment by fda77** [comment]
Tsss, über 10 Jahre hat das jeder hinbekommen und in letzter Zeit häufen sich die Beschwerden...

Es gibt einen optionalen Patch der load+save entfernt:
 https://github.com/Freetz-NG/freetz-ng/blob/master/make/host-tools/kconfig-host/patches/buttons/900-no_load_and_save_buttons.patch
 Als ich den aktivierte haben sich viele gebläht dass es extrem ganz wichtig wäre die config unter beliebigen Namen zu speichern
Ich aktivieren den mal als default

**Reply by tutti21** [reply]
https://github.com/Freetz-NG/freetz-ng/blob/master/README.md#requirements ich bin jetzt da nach gegangen. Ka ob man da vielleicht hinschreibt das der config name default bleiben muss da er sonst einfach die Voreinstellungen schreibt und ich die ausgewaehlten. Und jeder der sich damit schonmal auskennt kann dann ja mit der config was anderes machen aber fuer nen Anfaenger ware das glaub ich ganz gut dem unter die Arme zu greifen sonst war ich bestimmt nicht der letzte hier.

**Reply by fda77** [reply]
Ich selbst hab die config noch nie umbenannt. keine Ahnung warum die die es brauchen es brauchen...
Man kan es jedenfalls bald wieder nachlesen wenn einige nörgeln :)

**Reply by tutti21** [reply]
ich sag mal danke, fuer die Neuen gut fuer die einige veteranen vielleicht nicht so toll aber ich hoffe das die besser damit klar kommen als die andere Fraktion.

**Reply by fda77** [reply]
Weisst du, eigentlich ist überall so wo "kconfig" benutzt wird. Beim kernel, uclib+ng,openwrt usw.
Daher hatte ich auch den Patch gemacht, weil es einfach keinen Sinn macht der Datei einen anderen Namen zu geben

**Comment by fda77** [comment]
ohne dass man was im menuconfig auswählt gibt es kein load+save mehr: 4a51b92e4

**Reply by aus-sua** [reply]
zum Patch: s.u.

unabhängig davon:
Mir ist schon klar, daß man sich/Dir mit dem load/save auch noch zusätzliche "Probleme" macht, 
z.B. daß die Leute "ihre ach so tolle" aber eben ranzige, alte .config (ehemals FOS 04.67) von vor 
15 Jahren in die aktuelle FOS 08.02 rück-loaden...
Aber das war ja hier nicht der Fall.
Wenn irgend wo erst einmal steht: kopiere die alte .config (von 2009...) einfach in das neu ausgecheckte
Verzeichnis für die FOS 08.02, dann bringt ja auch das: "es gibt kein load/save" etwas...
Im Endeffekt ist das ja kconfig by design.

Erlaube load/save doch nur/wenigstens auf "Developer" und gut ist.
Für genau solche Fälle gab/gibt es ja die Gruppierung...

Zumindest die letzten Fälle von: "ich bin Anfänger und weiß es nicht" hätten sich da erledigt...

**Comment by aus-sua** [comment]
> ohne dass man was im menuconfig auswählt gibt es kein load+save mehr

Der Patch hat doch schon letztens nicht so wie beschrieben "gewirkt", weil er im 
Endeffekt wohl nur auf die .config testet?!
Beim ersten neu-auschecken in ein neues Verzeichnis ist die .config nicht 
vorhanden: ---> load+save sind doch da!
Beim "Exit" wird die .config geschrieben, beim nächsten "make menuconfig" sind 
dann "load+save" verschwunden und **bleiben** auch verschwunden, egal ob ich 
etwas auswähle oder nicht (ich kann ja nicht zwischenspeichern (oder liegt es 
an einer nicht vorhanden Vorinitialisierung/kconfig?)).
Und zusätzlich:
> Tsss, über 10 Jahre hat das jeder hinbekommen...

Ich erhöhe da 'mal auf:
Über 15 Jahre hat das jeder hinbekommen!
Warum also zusätzlich Arbeit reinstecken (bzw. s.u.)? 

Sorry, aber freetz/freetz-NG war noch **nie** narrensicher und wird es auch nie werden.
Leute, die nur eine "Waschanleitung" befolgen, ohne nach zu lesen und ohne mit zu denken
(oder auch "zu gut" denken), bekommen wohl immer irgend wann Probleme... 
(aka: Wer seinen roten Hipster-Stringtanga gleichzeitig mit seinem weißen Kaschmir-Pullover 
wäscht, heult doch dann auch nicht beim Waschmaschinen-Hersteller rum...)
Das Problem ist doch wohl eher die Vorbelegung mit der "7690er", keine Ahnung, 
ob "kconfig" da  etwas beim ersten Initialisieren braucht (?), ggf. kann man das ja 
schon weglassen?!
Aber warum denn nur wieder?

@tutti21
Wenn Dir in der Anleitung etwas fehlt, dann schreibe es verdammt noch einmal dort rein!
Hier wird um ein Problem herum gedockter und Arbeit hereingesteckt, daß mit einigen 
Worten  in der Doku und in der Hilfe von menuconfig dann wohl keines mehr wäre.
Du hast Dich bei github angemeldet, dann schaffst Du das auch! 
Also los! 
**Schreibe es dort rein!**

**Comment by fda77** [comment]
Ja, später ist mir das auch noch eingefallen und ich hab die Logik invertiert. Wie gesagt wollte ich die buttons eh schon lange weg haben
Wer Wert auf diese 2 buttons legt musss nun die Option aktivieren.
Mal schauen wem das nicht gefällt und was die Gründe sein werden
